app/src/main/javascript/bundles/experiment-page/src/ExperimentPageRouter.js

```diff
@@ -10,130 +10,169 @@ import SupplementaryInformationRoute from './SupplementaryInformationRoute'
+    for (let { type, key, component, optionsKey } of tabTypes) {
+        if (isThisTabType(tab, type)) {
+            const propValue = getNestedProperty(tab.props, key);
+            const optionsValue = optionsKey ? getNestedProperty(tab.props, optionsKey) : true;
+
+            if (isNonEmptyArray(propValue) && (!optionsKey || !isObjectEmpty(optionsValue))) {
+                tabTypeComponent.push({ [type]: component });
+                return tab.name;
+            }
+        }
+    }
+    return null;
+};
+
+const TopRibbon = ({tabs, routeProps}) => {
+    tabTypeComponent = [];
+    return <ul className={`tabs`}>
+        {
+            tabs.map((tab) => (
+                <li title={tab.name} key={tab.type} className={`tabs-title`}>
+                    <NavLink to={{
+                        pathname: `/${tab.type}`,
+                        search: routeProps.location.search,
+                        hash: routeProps.location.hash
+                    }}
+                             activeClassName={`active`}>
+                        { enableExperimentPageTab(tab) }
+                    </NavLink>
+                </li>))
+        }
+    </ul>
+}
 TopRibbon.propTypes = {
-  tabs: PropTypes.arrayOf(PropTypes.shape({
-    type: PropTypes.string.isRequired,
-    name: PropTypes.string.isRequired,
-    props: PropTypes.object
-  })).isRequired,
-  routeProps: PropTypes.shape(RoutePropTypes)
+    tabs: PropTypes.arrayOf(PropTypes.shape({
+        type: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+        props: PropTypes.object
+    })).isRequired,
+    routeProps: PropTypes.shape(RoutePropTypes)
 }
 
 
 const TabContent = ({type, tabProps, commonProps, routeProps}) => {
-  // Pass in the search from location
-  const Tab = tabTypeComponent[type]
+    // Pass in the search from location
+    const Tab = tabTypeComponent[type]
 
-  return (
-    Tab ? <Tab {...tabProps} {...commonProps} {...routeProps}/> : null
-  )
+    return (
+        Tab ? <Tab {...tabProps} {...commonProps} {...routeProps}/> : null
+    )
 }
 
 TabContent.propTypes = {
-  type: PropTypes.string.isRequired,
-  tabProps: PropTypes.object,
-  commonProps: PropTypes.shape(TabCommonPropTypes),
-  routeProps: PropTypes.shape(RoutePropTypes)
+    type: PropTypes.string.isRequired,
+    tabProps: PropTypes.object,
+    commonProps: PropTypes.shape(TabCommonPropTypes),
+    routeProps: PropTypes.shape(RoutePropTypes)
 }
 
 const RedirectWithSearchAndHash = (props) =>
-  <Redirect to={{ pathname: props.pathname, search: props.location.search, hash: props.location.hash}} />
+    <Redirect to={{pathname: props.pathname, search: props.location.search, hash: props.location.hash}}/>
 
 RedirectWithSearchAndHash.propTypes = {
-  pathname: PropTypes.string.isRequired,
-  location: PropTypes.shape({
-    search: PropTypes.string.isRequired,
-    hash: PropTypes.string.isRequired
-  }).isRequired
+    pathname: PropTypes.string.isRequired,
```

Comment:
The `proptypes` are all over this JS file.
As I already mentioned, I would externalise them to a separate file and import them into this one.
That would make this file more clear.
What do you think?

Reply:
I agree with you, I will try to do that.

Reply:
I don't know what you meant by `This one is fixed`. Sadly nothing has changed here.